### DISCUSSION TITLE
how to plot the averaged ERPs of ROI?

### DISCUSSION THREAD

**Discussion by ghzhlchen** [bnicenboim/eeguana#240] [Q&A] [answered]
My scripts are problematic. But I can't figure it out quickly? Can you help me? Or share your scripts of plotting it?

explt %>% 
  eeg_transmute(
    ROI = chs_mean(across(c("C3", "Cz", "C4", "CP1", "CP2"), na.rm = TRUE))
  ) %>% 
  ggplot(aes(x = .time, y = .value)) +
  geom_line(alpha = 0.1, aes(group = .id, color = condition)) +
  stat_summary(
    fun = "mean", geom = "line", alpha = 1, size = 1,
    aes(color = condition)
  ) +
  facet_wrap(~ .condition) +
  theme(legend.position = "bottom")

**Comment by bnicenboim** [comment]
Dear [ghzhlchen](https://github.com/ghzhlchen), 
As I mentioned last time, it's just impossible to help you without a fully reproducible example with data that I can test myself.

FYI, I'm right now in the middle of mathpsych/Iccm conference organization, which starts tomorrow., so I won't be checking the forum until it finishes.

**Comment by ghzhlchen** [comment]
Dear Bruno, 

Thank you for your reply. Sorry again. 
My problem exists also in the dataset that I used in another question.
https://drive.google.com/file/d/1xwxwNAgyjF_RuOg88tOPYVnTOJNRA8At/view?usp=drive_link
We can just use this dataset. 

```
# Load multiple subjects using purrr::map, extracting subject IDs from file names.
COM_list <- purrr::map(list.files("./", "vhdr"), ~
read_vhdr(.x))
# Bind all the eeg_lsts into a large one:
COM <- eeg_bind(COM_list)

#add condition and trial to .segment
events_table <- COM$.events

# generate .trial and .condition columns
result <- events_table %>%
  group_by(.id) %>%
  mutate(
    .trial = .description[which(.type == "Stimulus")[1]],
    .condition = .description[which(.type == "Stimulus")[2]]
  ) %>%
  slice(1) %>%
  select(.id, .trial, .condition) %>%
  ungroup()

COM <- COM %>%
  left_join(., result %>% select(.id, .trial, .condition), by = ".id")

#left join RC_type
# read the coding table
RCtype_coding <- read.csv("RCtype_coding.csv")
RCtype_coding

# left join the coding table to the DEM list and rename type_exclude_coding
COM <- left_join(COM, select(RCtype_coding, .trial, type_exclude_coding), by = ".trial") %>%
  rename(RC_type = type_exclude_coding)

#grand-averaged erps
ERP_COM <- COM %>%
  filter(RC_type != "EX") %>%   # Exclude rows where type_exclude_coding is "EX"
  eeg_group_by(.sample, RC_type) %>%
  eeg_summarize(across_ch(mean, na.rm = TRUE))

COM_plot <- ERP_COM %>%
  ggplot(aes(x = .time, y = .value)) +  # i removed the addition of 1600 ms 
  geom_line(aes(color = RC_type)) +
  facet_wrap(~.key) +
  theme(legend.position = "bottom") +
  ggtitle("ERPs for SR and OR on COMs") +
   # Adjust the limits as needed
  # scale_x_continuous(limits = c(-0.2, 1.0)) + # this now doesn't suit the data
  theme_eeguana()
COM_plot %>% plot_in_layout()

#until here the scripts are fine, I don't know how to plot the averaged erps for ROI
#Here are my scripts 
COM %>%
eeg_transmute(
ROI = chs_mean(across(c("C3", "Cz", "C4", "CP1", "CP2"), na.rm = TRUE))
) %>%
ggplot(aes(x = .time, y = .value)) +
geom_line(alpha = 0.1, aes(group = .id, color = RC_type)) +
stat_summary(
fun = "mean", geom = "line", alpha = 1, size = 1,
aes(color = RC_type)
) +
theme(legend.position = "bottom")
```

**Reply by stonekate** [reply]
Hey @ghzhlchen (and @bnicenboim), I can take a look tomorrow

**Reply by ghzhlchen** [reply]
Thank you!!!

**Reply by ghzhlchen** [reply, marked as answer]
Hi! I think I found a way to quickly fix my problem. That is to turn the list to a dataframe and then use ggplot() to plot. 
The error information was: 
Error in `fortify()`:
! `data` must be a <data.frame>, or an object coercible by `fortify()`, not a <uneval> object.
ℹ Did you accidentally pass `aes()` to the `data` argument?
Backtrace:
 1. ggplot2::ggplot(aes(x = .time, y = .value))
 2. ggplot2:::ggplot.default(aes(x = .time, y = .value))
 4. ggplot2:::fortify.default(data, ...)

**Reply by stonekate** [reply]
Hey, sorry, I didn't get to this but yes indeed it does need to be converted to a dataframe first... good solve! :D
